tools/hitmap/hitmap.xml

```diff
@@ -0,0 +1,297 @@
+
+<tool id="hitmap" name="HiT-MaP" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" license="MIT">
+    <description>High-resolution Maldi-imaging Proteomics Analysis</description>
+
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+
+    <edam_topics>
+        <edam_topic>topic_0121</edam_topic>
+    </edam_topics>
+
+    <edam_operations>
+        <edam_operation>operation_3630</edam_operation>
+    </edam_operations>
+
+    <requirements>
+        <container type="docker">mashuoa/hitmap@sha256:97f2f853362b56be6c2b0fb1507e0288667e7fb6ad0eda3ff1d761dd72dfaad0</container>   
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir expdata &&
+        cp '${file_inputs.imzml_file}' expdata/sample.imzML &&
+        cp '${file_inputs.ibd_file}' expdata/sample.ibd &&
+        cp '${file_inputs.fasta_db}' expdata/database.fasta &&
+        Rscript '${hitmap_script}' &&
+        python3 $__tool_directory__/hitmap_gen_html.py expdata &&
+        zip -r hitmap_output.zip expdata
+    ]]></command>
+
+    <configfiles>   
+        <expand macro="hitmap_config_script" />
+    </configfiles>
+
+    <inputs>
+        <section name="file_inputs" title="File Inputs" expanded="True">
+            <param  name="imzml_file" 
```

Comment:
The Galaxy datatype imzml is a composite datatype, which means it contains 2 files. You can see how this can be used with the Cardinal tools.

https://github.com/galaxyproteomics/tools-galaxyp/blob/master/tools/cardinal/macros.xml#L18

Reply:
I see.  I originally tried to do it that way, but couldn't work out how to upload two files (.imzml + .ibd) together as a composite type in the galaxy uploader. 

I just went with the approach that the users upload the imzml and ibd as seperate files into their history, then select those when performing the run. 

What do you think about that approach? 

Also thank you again so much for looking over these submissions and helping me out. I really appreciate it!

Reply:
> What do you think about that approach?

I think its can result in more errors from the user side, picking the wrong combination of files etc.
Do you know why this did not work for you? Can you try this on EU. I think its working on EU.

![grafik](https://user-images.githubusercontent.com/469983/134874267-696d1c38-1fea-4c0d-9c0f-65b719c255d7.png)

Reply:
Ohhhh I see! 
Yes this is working fine, I just wasn't aware of that option. Very neat! 
Added to the next commit